cumulus/client/cirrus-executor/src/bundle_processor.rs

```diff
@@ -185,6 +193,24 @@ where
         shuffling_seed: Randomness,
         maybe_new_runtime: Option<Cow<'static, [u8]>>,
     ) -> Result<(), sp_blockchain::Error> {
+        {
+            let mut pending_unsigned = self.pending_unsigned.lock();
```

Comment:
Hmm, I intended to convey that the message producers(bundle_processor, gossiped_message_handler) can easily build their own additional buffer on top of the global one based on their priority, they're considered to be equivalently important for now though.

`pending_unsigned` can be replaced with checking if there is still some capacity in the channel and only continues when the channel still has some capacity, which is kind of conservative as not every `process_bundles` will be needed to submit this unsigned extrinsic. I have no strong opinion.

Reply:
Why would they need to have their own buffer if they can't push messages into the global one anyway? Why not have multiple separate bounded channels instead of this one then? It seems to be overengineered and is hard to verify correctness of.